File: src/main/java/io/github/boogiemonster1o1/cartses/client/MinecartWithRedstoneLampEntityRenderer.java
package io.github.boogiemonster1o1.cartses.client;

import io.github.boogiemonster1o1.cartses.entity.cart.MinecartWithRedstoneLampEntity;

import net.minecraft.block.Blocks;
import net.minecraft.block.RedstoneLampBlock;
import net.minecraft.client.render.entity.EntityRenderDispatcher;
import net.minecraft.client.render.entity.MinecartEntityRenderer;
import net.minecraft.util.math.BlockPos;

public class MinecartWithRedstoneLampEntityRenderer extends MinecartEntityRenderer<MinecartWithRedstoneLampEntity> {
	public MinecartWithRedstoneLampEntityRenderer(EntityRenderDispatcher dispatcher) {
		super(dispatcher);
	}

	@Override
	protected int method_27950(MinecartWithRedstoneLampEntity entity, BlockPos blockPos) {
		return entity.isLit() ? Blocks.REDSTONE_LAMP.getDefaultState().with(RedstoneLampBlock.LIT, true).getLuminance() : super.method_27950(entity, blockPos);
	}

	@Override
	protected int getBlockLight(MinecartWithRedstoneLampEntity entity, BlockPos blockPos) {
		return entity.isLit() ? Blocks.REDSTONE_LAMP.getDefaultState().with(RedstoneLampBlock.LIT, true).getLuminance() : super.getBlockLight(entity, blockPos);
	}
}

File: src/main/scala/io/github/boogiemonster1o1/cartses/client/MinecartWithRedstoneLampEntityRenderer.scala
package io.github.boogiemonster1o1.cartses.client

import io.github.boogiemonster1o1.cartses.entity.cart.MinecartWithRedstoneLampEntity
import net.fabricmc.api.{EnvType, Environment}
import net.minecraft.block.{Blocks, RedstoneLampBlock}
import net.minecraft.client.render.entity.{EntityRenderDispatcher, MinecartEntityRenderer}
import net.minecraft.util.math.BlockPos


@Environment(EnvType.CLIENT) class MinecartWithRedstoneLampEntityRenderer(val dispatcher: EntityRenderDispatcher) extends MinecartEntityRenderer[MinecartWithRedstoneLampEntity](dispatcher) {
	override protected def method_27950(entity: MinecartWithRedstoneLampEntity, blockPos: BlockPos): Int = if (entity.isLit) Blocks.REDSTONE_LAMP.getDefaultState.`with`(RedstoneLampBlock.LIT, true).getLuminance else super.method_27950(entity, blockPos)

	override protected def getBlockLight(entity: MinecartWithRedstoneLampEntity, blockPos: BlockPos): Int = if (entity.isLit) Blocks.REDSTONE_LAMP.getDefaultState.`with`(RedstoneLampBlock.LIT, true).getLuminance else super.getBlockLight(entity, blockPos)
}
